pymc/distributions/mixture.py

```diff
@@ -27,28 +27,31 @@
 from pymc.distributions.continuous import Normal, get_tau_sigma
 from pymc.distributions.dist_math import check_parameters
 from pymc.distributions.distribution import (
-    Discrete,
-    Distribution,
     SymbolicDistribution,
     _get_moment,
     get_moment,
 )
 from pymc.distributions.logprob import logcdf, logp
 from pymc.distributions.shape_utils import to_tuple
 from pymc.util import check_dist_not_registered
-from pymc.vartypes import discrete_types
+from pymc.vartypes import continuous_types, discrete_types
 
 __all__ = ["Mixture", "NormalMixture"]
 
 
-def all_discrete(comp_dists):
+def all_same_type(comp_dists):
     """
-    Determine if all distributions in comp_dists are discrete
+    Determine if the distributions in comp_dists are all discrete or continuous
     """
-    if isinstance(comp_dists, Distribution):
-        return isinstance(comp_dists, Discrete)
+
+    if len(comp_dists) == 1:
```

Comment:
Can't we just return True immediately?

Reply:
I guess so. Are there cases where this could be an issue?

Reply:
Can't think of any